external/fv3fit/fv3fit/keras/_models/shared/pure_keras.py

```diff
@@ -175,3 +176,80 @@ def input_sensitivity(self, stacked_sample: xr.Dataset) -> InputSensitivity:
         std_factors = {name: np.std(data, axis=0) for name, data in data_dict.items()}
         jacobians_std = nondimensionalize_jacobians(jacobians, std_factors)
         return InputSensitivity(jacobians=jacobians_std)
+
+
+@io.register("all-keras-dict")
+class PureKerasDictPredictor(Predictor):
+    """Save a keras model encoding the input/output names into a saved_model
+    artifact
+
+    The output names of the keras model are assumed to be correct. To do this
+    requires including the following logic when building the model::
+
+        out_with_correct_name = tf.keras.layers.Lambda(
+            lambda x: x, name="desired_name")
+
+    Note:
+        This class relies on PureKerasModel to implement Mhe Predictor methods.
+        It might be worth refactoring the implementation here instead since the
+        dict-in/out format has more structure.
+
+    """
+
+    _MODEL_FILENAME = "model.tf"
+    _CONFIG_FILENAME = "config.yaml"
+
+    def __init__(self, model: tf.keras.Model, passthrough: Any = None):
+        """Initialize the predictor.
+
+        Args:
+            model: a keras model. Tuple in/out models are okay, but will be cast
+                to a dict-output model before saving.
+            passthru: any data! Is NOT serialized, but can be
```

Comment:
suggestion: snake_case? I'm also fine with saying passthrough is one word.
```suggestion
            passthrough: any data! Is NOT serialized, but can be
```